src/d3a/models/area/markets.py

```diff
@@ -15,84 +15,158 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from pendulum import DateTime # noqa
-from typing import Dict  # noqa
-
-from d3a.models.market import GridFee
-from d3a.models.market.two_sided import TwoSidedMarket
-from d3a.models.market.one_sided import OneSidedMarket
-from d3a.models.market.balancing import BalancingMarket
-from d3a.models.market import Market # noqa
-from d3a_interface.constants_limits import ConstSettings
 from collections import OrderedDict
+from typing import Dict, List, TYPE_CHECKING
+
+from d3a import constants
 from d3a.d3a_core.util import is_timeslot_in_simulation_duration
-import d3a.constants
+from d3a.models.market import GridFee, Market
+from d3a.models.market.balancing import BalancingMarket
+from d3a.models.market.one_sided import OneSidedMarket
+from d3a.models.market.two_sided import TwoSidedMarket
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from pendulum import DateTime
+
+if TYPE_CHECKING:
+    from d3a.models.area import Area
```

Comment:
I think this is not needed if you use `"Area"` as a string, as you're doing below.

Reply:
Actually no, flake will complain if I dont do that.

Reply:
Weird, I didn't know that. Thanks!